cmd/server/main.go

```diff
@@ -2,26 +2,84 @@ package main
 
 import (
 	"encoding/json"
+	"flag"
 	"fmt"
 	"io"
 	"log"
+	"net"
 	"net/http"
+	"os"
 	"time"
 
+	"github.com/joho/godotenv"
+
 	"github.com/benchttp/engine/internal/configparse"
 	"github.com/benchttp/engine/runner"
 )
 
 const (
-	port = "8080"
+	ReadySignal = "READY"
+)
+
+var (
+	stdout = log.New(os.Stdout, "", 0)
+	stderr = log.New(os.Stderr, "", log.LstdFlags)
 )
 
 func main() {
-	addr := ":" + port
-	fmt.Println("http://localhost" + addr)
+	useAnyPort := flag.Bool("any-port", true, "use any available port allocated by the os")
+	flag.Parse()
+
+	var p string
+	if !*useAnyPort {
+		err := godotenv.Load("./.env.development")
+		if err != nil {
+			log.Println(err)
+		}
+		p = os.Getenv("VITE_ENGINE_PORT")
+	} else {
+		p = "0"
+	}
+
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:"+p)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	l, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	port := l.Addr().(*net.TCPAddr).Port
+
+	//#nosec G112 -- Potential Slowloris Attack because
+	// ReadHeaderTimeout is not configured in the http.Server.
+	// Ignored because the end user runs both the client and
+	// the server on their machine, which makes it irrelevant.
+	s := &http.Server{
+		Addr:    ":" + fmt.Sprint(port),
+		Handler: http.HandlerFunc(handleStream),
+	}
+
+	readyChan := make(chan struct{}, 1)
+	closeChan := make(chan error, 1)
+
+	go func() {
+		readyChan <- struct{}{}
+		closeChan <- (s.Serve(l))
+	}()
+
+	<-readyChan
+	// From now we communicate with the parent process via stdout and stderr.
+	// Notify server is ready.
+	stdout.Println(sprintReadySignal(port))
+
+	stderr.Fatal(<-closeChan)
+}
 
-	//#nosec G114 -- Ignored for convenience
-	log.Fatal(http.ListenAndServe(addr, http.HandlerFunc(handleStream)))
+func sprintReadySignal(port int) string {
```

Comment:
Calling it `sprint` because it returns a string / because it uses `sprintf` internally is a bit weird 😅 
```suggestion
func readyString(port int) string {
```